GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m

```diff
@@ -91,4 +110,64 @@ - (void)testLegacyCoding {
 }
 #endif // TARGET_OS_IOS || TARGET_OS_MACCATALYST
 
+- (void) testUpdateAuthState {
+  GIDGoogleUser *user = [self googleUser];
+  XCTAssertEqualObjects(user.accessToken.tokenString, kAccessToken);
+  [self assertDate:user.accessToken.expirationDate equalTime:_accessTokenExpireTime];
+  XCTAssertEqualObjects(user.idToken.tokenString, [self idToken]);
+  [self assertDate:user.idToken.expirationDate equalTime:_idTokenExpireTime];
+  
+  OIDAuthState *newAuthState = [self newAuthState];
+  [user updateAuthState:newAuthState profileData:nil];
+  
+  XCTAssertEqualObjects(user.accessToken.tokenString, kNewAccessToken);
+  [self assertDate:user.accessToken.expirationDate equalTime:kNewExpireTime];
+  XCTAssertEqualObjects(user.idToken.tokenString, [self idTokenNew]);
+  [self assertDate:user.idToken.expirationDate equalTime:kNewExpireTime2];
+}
+
+#pragma mark - Helpers
+
+- (GIDGoogleUser *)googleUser {
+  NSString *idToken = [self idToken];
+  NSNumber *accessTokenExpiresIn =
+      @(_accessTokenExpireTime - [[NSDate date] timeIntervalSinceReferenceDate]);
+  OIDTokenRequest *tokenRequest =
+      [OIDTokenRequest testInstanceWithAdditionalParameters:nil];
+  OIDTokenResponse *tokenResponse =
+      [OIDTokenResponse testInstanceWithIDToken:idToken
+                                    accessToken:kAccessToken
+                                      expiresIn:accessTokenExpiresIn
+                                   tokenRequest:tokenRequest];
+  return [[GIDGoogleUser alloc]
+          initWithAuthState:[OIDAuthState testInstanceWithTokenResponse:tokenResponse]
+                profileData:nil];
+}
+
+- (NSString *)idToken {
```

Comment:
This and the getter below basically do the same thing with a different expiration time, right? Do you need these methods? If you're using these methods as a kind of named storage, consider using a property (or ivar) for each to make things clearer.